src/nxt_fs.c

```diff
@@ -58,7 +59,7 @@ nxt_fs_mkdir_dirname(const u_char *path, mode_t mode)
     ret = NXT_OK;
 
     ptr = strrchr(dir, '/');
-    if (nxt_slow_path(ptr == NULL)) {
+    if (ptr == dir || nxt_slow_path(ptr == NULL)) {
```

Comment:
I wonder if `path == "/"` should constitute an error?, as that's clearly not what you'd want.

Reply:
One may legitimatelly want to `--control=/unit.control.sock`, if they don't care about FHS and having an organized system.

We probably agree that they should be punished by such an offense, but misbehaving, or even prohibiting, is probably not the right punishment